regression/ansi-c/always_inline1/main.c

```diff
@@ -0,0 +1,37 @@
+// this is a GCC extension
+
+#ifdef __GNUC__
+static inline __attribute__((always_inline)) _Bool
```

Comment:
Could you minimise this test-case so it's clearer which parts are crucial to the test and which is unrelated stuff the kernel's RCU code happens to do? If you want to keep the real-world example you could include this as a block comment, then use a simplified version for the actual test?

Reply:
Actually everything is necessary here, but I'll add comments to explain why/what it is testing.